agent/src/main/scala/za/co/absa/atum/agent/dispatcher/HttpDispatcher.scala

```diff
@@ -47,19 +48,19 @@ class HttpDispatcher(config: Config) extends Dispatcher(config: Config) with Log
   override protected[agent] def createPartitioning(partitioning: PartitioningSubmitDTO): AtumContextDTO = {
     val request = commonAtumRequest
       .post(createPartitioningEndpoint)
-      .body(SerializationUtils.asJson(partitioning))
+      .body(partitioning.asJson.noSpaces)
```

Comment:
I find this a bit inconsistent. If you intent to have the `JsonUtils` class with `fromJsonString` method, then I think you should use the `asJsonString` method instead of calling `asJson.noSpaces`. And this way it should be done all over the code base.

Reply:
Alternatively we could use extension methods.
```scala
object JsonSyntaxExtensions {
  implicit class JsonSerializationSyntax[T: Encoder](val obj: T) extends AnyVal {
    def asJsonString: String = obj.asJson.noSpaces
    def asJsonStringPretty: String = obj.asJson.spaces2
  }
  implicit class JsonDeserializationSyntax[T: Decoder](val jsonStr: String) extends AnyVal {
    def fromJson: T = {
      decode[T](jsonStr) match {
        case Right(value) => value
        case Left(error) => throw new RuntimeException(s"Failed to decode JSON: $error")
      }
    }
  }
}

val person = Person("Alice", 30)
 
// Convert a Person instance to a compact JSON string
val jsonString = person.asJsonString
println(jsonString) // {"name":"Alice","age":30}
 
// Convert a Person instance to a pretty JSON string
val jsonPrettyString = person.asJsonStringPretty

val jsonStr = """{"name":"Bob","age":25}"""
 
// Convert a JSON string back to a Person instance
val personFromJson = jsonStr.fromJson[Person]
```

Reply:
This I tried to do, but it keeps saying that `JsonSerializationSyntax` and `JsonDeserializationSyntax` must have one input and it must be val. So, I switched back to our old implementation.

Reply:
Working solution
```scala
object JsonSyntaxExtensions {
  implicit class JsonSerializationSyntax[T: Encoder](val obj: T)  {
    def asJsonString: String = obj.asJson.noSpaces
    def asJsonStringPretty: String = obj.asJson.spaces2
  }
  implicit class JsonDeserializationSyntax(val jsonStr: String) {
    def as[T: Decoder]: T = {
      decode[T](jsonStr) match {
        case Right(value) => value
        case Left(error) => throw new RuntimeException(s"Failed to decode JSON: $error")
      }
    }
  }
}
```